lib/travis/logs/services/process_log_part.rb

```diff
@@ -107,6 +108,11 @@ def run
         end
         alias find_or_create_log log_id
 
+        private def normalize_number
+          return unless payload['number'] == 'last'
+          payload['number'] = database.max_log_part_number_for_log(log_id)
```

Comment:
In what case is this called? looks like there's a good chance for race conditions here if there are still log parts in a queue somewhere.